src/rpc/rawtransaction.cpp

```diff
@@ -179,6 +179,7 @@ UniValue getrawtransaction(const JSONRPCRequest& request)
     if (!GetTransaction(hash, tx, Params().GetConsensus(), hash_block, true, blockindex)) {
         std::string errmsg;
         if (blockindex) {
+            LOCK(cs_main);
```

Comment:
Note: it looks like adding this lock here is just a bugfix, and not directly related to the other changes.